### ISSUE TITLE
Flexibility for converting data with different number of bins per ping or frequency

### ISSUE THREAD

**Issue by leewujung** [OSOceanAcoustics/echopype#24]
Potential solutions:
- Saving echo data from different frequency channels into different groups when the `range_bin` dimension is different?
- Padding ping-by-ping echo data with NaNs if `range_bin` is different for each ping?

This issue is from #6.

**Comment by leewujung** [comment]
Also there are `.raw` files that actually contain pings with different number of range bins. This probably happened because the operator changed the collection of data range in the middle of a recording file. Maybe the groups of pings that contain different range bins should be in different netCDF files so that it is explicit??

**Comment by leewujung** [comment]
Adding original comment from #9 here:

If any one of these parameters change in the middle of a .raw or .01a file, writing to the netCDF should stop, the netCFD should be saved, and a new netCDF should be opened and written to.
- Power
- Pulse duration
- Sample unit (ping rate)
- Receiver bandwidth
- Sound speed
- Start/Stop of a file

**Comment by leewujung** [comment]
This issue for EK60 files (in both convert and model modules) is resolved in #91. Need to find corresponding AZFP files to update convertAZFP. The model module methods should just work for AZFP data.

**Comment by leewujung** [comment]
This went through a few rounds of changes but the solution is stable now. This can be closed once class-redesign is merged into master.

**Comment by leewujung** [comment]
Closing this with #298.
